demo-circuit/src/naive_threshold_sig/mod.rs

```diff
@@ -230,11 +276,24 @@ impl<F: PrimeField> ConstraintSynthesizer<FieldElement> for NaiveTresholdSignatu
             },
         )?;
 
+        // Alloc custom_fields and enforce their hash, if they are present
+        let custom_fields_hash_g = if let Some(custom_fields) = self.custom_fields.clone() {
```

Comment:
If I understand correctly this is a circuit for the BFT Withdrawal Certificate proof. Note that according to specification BFT Wcert contains 2 custom fields: `scb_new_mst_root` and `submitter_key`. And `submitter_key` should not be included in the message signed by the servers, because at the moment of signing the submitter isn't known yet. Here it seems all custom fields constitutes message to sign.

Reply:
Yes, due to an issue in MC we were forced to split a single field element, the scb_new_mst_root, into 2.
submitter_key is not part of the custom fields as I understood.

Reply:
Then, where is the submitter key? Or this circuit is not fully implemented yet?

Reply:
submitter_pubkey is not used in the circuit, but all custom fields of the certificate are allocated and hashed inside the circuit. I guess it is correctly passed in the SDK, right @i-Alex ?

Reply:
Yes, everything you described is correct. Some custom fields should be signed (scb_new_mst_root) and some other should not (submitter_key). At some point this should be implemented (at least for full Latus sidechain). But for the BFT we will not have submitter_key, did I understand correctly?